From b8f61f5bf3ae496cad9ab27d4b94f6c45dae9d10 Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Mon, 28 Oct 2019 17:59:52 +0800
Subject: [PATCH] drm/rockchip: vop: fix multile area win state error

the supported multile area win gate only can be closed when area0 is
disabled.

Change-Id: Idd447bb7c2157587456391c1abeeefc9a5d61bbd
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
index 2cf1ae300828..688296788ed7 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
@@ -520,7 +520,8 @@ static void vop_win_disable(struct vop *vop, struct vop_win *win)
 	}
 
 	VOP_WIN_SET(vop, win, enable, 0);
-	VOP_WIN_SET(vop, win, gate, 0);
+	if (win->area_id == 0)
+		VOP_WIN_SET(vop, win, gate, 0);
 }
 
 static void vop_disable_allwin(struct vop *vop)
-- 
2.35.3

